ako-gateway-api/lib/control_config.go

```diff
@@ -0,0 +1,178 @@
+	gwApi "sigs.k8s.io/gateway-api/pkg/client/clientset/versioned"
+	gwApiInformer "sigs.k8s.io/gateway-api/pkg/client/informers/externalversions/apis/v1beta1"
+)
+
+type gwApiInformers struct {
+	GatewayInformer      gwApiInformer.GatewayInformer
+	GatewayClassInformer gwApiInformer.GatewayClassInformer
+}
+
+// akoControlConfig struct is intended to store all AKO related global
+// variables, that are set as part of AKO bootup. This is a store of client-sets,
+// informers, config parameters, and internally computed static configurations.
+
+type akoControlConfig struct {
+
+	// client-set and informer for Gateway API.
+	gwApiCS        gwApi.Interface
+	gwApiInformers *gwApiInformers
+
+	// akoEventRecorder is used to store record.akoEventRecorder
+	// that allows AKO to broadcast kubernetes Events.
+	akoEventRecorder *utils.EventRecorder
+
+	// akoPodObjectMeta holds AKO Pod ObjectMeta information
+	akoPodObjectMeta *metav1.ObjectMeta
+
+	// licenseType holds the default license tier which would be used by new Clouds. Enum options - ENTERPRISE_16, ENTERPRISE, ENTERPRISE_18, BASIC, ESSENTIALS.
+	licenseType string
+
+	// controllerVersion stores the version of the controller to
+	// which AKO is communicating with
+	controllerVersion string
+}
+
+var akoControlConfigInstance *akoControlConfig
+
+func AKOControlConfig() *akoControlConfig {
+	if akoControlConfigInstance == nil {
+		akoControlConfigInstance = &akoControlConfig{
+			controllerVersion: initControllerVersion(),
+		}
+	}
+	return akoControlConfigInstance
+}
+
+func (c *akoControlConfig) SetGatewayAPIClientset(cs gwApi.Interface) {
+	c.gwApiCS = cs
+}
+
+func (c *akoControlConfig) GatewayAPIClientset() gwApi.Interface {
+	return c.gwApiCS
+}
+
+func (c *akoControlConfig) SetGatewayApiInformers(i *gwApiInformers) {
+	c.gwApiInformers = i
+}
+
+func (c *akoControlConfig) GatewayApiInformers() *gwApiInformers {
+	return c.gwApiInformers
+}
+
+func (c *akoControlConfig) ControllerVersion() string {
+	return c.controllerVersion
+}
+
+func (c *akoControlConfig) SetControllerVersion(v string) {
+	c.controllerVersion = v
+}
+
+func initControllerVersion() string {
+	version := os.Getenv("CTRL_VERSION")
+	if version == "" {
+		return version
+	}
+
+	// Ensure that the controllerVersion is less than the supported Avi maxVersion and more than minVersion.
+	if lib.CompareVersions(version, ">", lib.GetAviMaxSupportedVersion()) {
+		utils.AviLog.Infof("Setting the client version to AVI Max supported version %s", lib.GetAviMaxSupportedVersion())
+		version = lib.GetAviMaxSupportedVersion()
+		return version
+	}
+
+	if lib.CompareVersions(version, "<", lib.GetAviMinSupportedVersion()) {
+		AKOControlConfig().PodEventf(
+			corev1.EventTypeWarning,
+			lib.AKOShutdown, "AKO is running with unsupported Avi version %s",
+			version,
+		)
+		utils.AviLog.Fatalf("AKO is not supported for the Avi version %s, Avi must be %s or more", version, lib.GetAviMinSupportedVersion())
+	}
+	return ""
+}
```

Comment:
This is not required in the Gateway pod since AKO is already doing this and will make the whole pod in `CrashLoop` state.